src/lib.rs

```diff
@@ -336,6 +336,15 @@ impl OctocrabBuilder {
                 AuthState::None
             }
             Auth::App(app_auth) => AuthState::App(app_auth),
+            Auth::OAuth(device) => {
```

Comment:
I'm currently testing the API for this PR for a project, and I can't find an obvious way to actually set this value.

[`OctocrabBuilder`](https://docs.rs/octocrab/latest/octocrab/struct.OctocrabBuilder.html) has `app`, `basic_auth`, and `personal_token` methods that set the other values for `auth`, but this PR doesn't seem to provide a way to set `auth` on the builder instance.